engine/mongo-storage.js

```diff
@@ -92,19 +88,19 @@ function MongoStorage(db) {
                     userData.variables = escapeNames(userData.variables)
                     callback(userData, false)
                 } else if (result.length === 0) {
-                    createUser(id, messenger, callback)
+                    createUser(c.userId, c.messengerApi.messenger, callback)
                 } else {
                     throw new Error("Many documents by the same ID: " + id + ": " + result.join(", "))
                 }
             }
         )
     }
 
-    this.saveUserData = (id, messenger, stack, variables, globalInputHandlers, callback) => {
+    this.saveUserData = (userData, stack, variables, globalInputHandlers, callback) => {
```

Comment:
This method is named `saveUserData`, you call a new parameter `userData`, what impression does this make on the reader?

Reply:
Also, some places you rely on the existing notion of context (`c`), and some places it's this `userData`, this is too many concepts for such a small thing